src/System.Console/src/System/ConsoleKeyInfo.cs

```diff
@@ -70,7 +70,11 @@ public bool Equals(ConsoleKeyInfo obj)
 
         public override int GetHashCode()
         {
-            return (int)_keyChar | (int)_mods;
+            // For all typical cases we can fit all bits losslessly into the hash code:
+            // _keyChar could be any 16-bit value (though is most commonly ASCII). Use 16 bits.
+            // _key can theoretically be 32-bit, but only has enum defined values up to 255.  Use 8 bits.
+            // _mods can also theoretically be 32-bit, but only has enum defined values up to 4. Use 8 bits.
+            return _keyChar | ((int)_key << 16) | ((int)_mods << 24);
```

Comment:
> _key is the scan code and 254 is actually used, it's OemClear

I'm not following.  It's effectively taking 8-bits from _key (though still factoring in more if they exist).  254 would fit within that, e.g. 
```C#
char keyChar = (char)0x1234;
ConsoleKey key = ConsoleKey.OemClear;
ConsoleModifiers mods = ConsoleModifiers.Alt | ConsoleModifiers.Control | ConsoleModifiers.Shift;
int hash = keyChar | ((int)key << 16) | ((int)mods << 24);
Console.WriteLine(hash.ToString("X"));
```
prints out:
```
7FE1234
```
What am I not understanding?